01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+  ggplot(aes(x = share_households)) +
+  geom_histogram(color = "blue", fill = alpha("blue", 0.3), binwidth = 0.02) +
+  facet_wrap(. ~ race_ethnicity) +
+  labs(
+    title = "Share of Households by Race and Ethnicity",
+    subtitle = "Place",
+    y = "Count",
+    x = "Share of Households"
+  )
+
+skimr::skim(share_households_place)
+summary(share_households_place)
+```
+
+```{r}
+#| label: check-share-households-place-age
+
+stopifnot(min(share_households_place_subgroup$share_households) >= 0)
+stopifnot(max(share_households_place_subgroup$share_households) <= 1)
+
+share_households_place_subgroup |>
+  ggplot(aes(x = share_households)) +
+  geom_histogram(color = "blue", fill = alpha("blue", 0.3), binwidth = 0.02) +
+  facet_wrap(subgroup ~ race_ethnicity, ncol = 2) +
+  labs(
+    title = "Share of Households by Race and Ethnicity",
+    subtitle = "Place",
+    y = "Count",
+    x = "Share of Households"
+  )
+
+skimr::skim(share_households_place_subgroup)
+summary(share_households_place_subgroup)
+```
+
+:::
+
+### Housing Value Share
+
+We keep observations for homeowners only in order to calculate total aggregated housing wealth for homeowners for each race and ethnic group divided by the total aggregated housing value
+
+```{r}
+#| label: total-house-values-by-race
+
+house_value_by_race <- function(data, ...) {
+  data |>
+    mutate(ownership = as_factor(ownership)) |>
+    filter(ownership == "Owned or being bought (loan)") |>
+    summarise(
+      house_value = sum(valueh * hhwt),
+      home_owners = n(),
+      .by = c(year, state, race_ethnicity, ...)
+    )
+}
+
+house_value_county_subgroup <- house_value_by_race(households_county, county, subgroup)
+house_value_county <- house_value_by_race(households_county, county)
+
+
+house_value_place_subgroup <- house_value_by_race(households_place, place, subgroup)
+house_value_place <- house_value_by_race(households_place, place)
+```
+
+
+The metrics for the ACS indices are based on:
+- (1) sample size of each race and ethnic household at the county/place and 
+- (2) the number of observations that comes from the specific county/place calculated from the PUMA-county/place crosswalk.
+
+Therefore, the data quality is poor for counties/places with a extremely small number of households (N). To assess the data quality of counties/places with larger N, we need to consider what percent of data for the county/place actually came from the county/place itself.
+
+In particular,
+
+-   Data quality is good (`1`): N is greater than 30 and more than 75% of observations are from the county/place.
+-   Data quality is marginal (`2`): N is greater than 30 but only 35% to 75% of observations are from the county/place.
+-   Data quality is poor (`3`): N is less than 30 **or** if less than 35% of observations are from the county/place
+
+```{r}
+#| label: quality-check-flag-functions
+
+construct_quality_flag <- function(x, y) {
+  case_when(
+    x < 30 ~ 3,
+    x >= 30 & y < 0.35 ~ 3,
```

Comment:
The text above this suggests that line 485 should be y < 0.35 only -- since it uses "or". If you mean "Data quality is poor (`3`): N is less than 30 **or** if **there are more than 30 observations but** less than 35% of observations are from the county/place" (as in the code as written is accurate)-- I would update the text above this code chunk.

Reply:
Thanks for catching this! The code indeed did not match what the text said, but the text was correct so I edited accordingly. I think this bug is part of the reason why there were missing values in the data quality flag.

Reply:
One thing I noticed though is that the vast majority of flag values is 3. This is because currently `N` refers to homeowners in a county/place instead of population. This is pretty strict and causes more observations to have a data quality flag value of 3 compared to the geographic_allocation_quality variable in the puma county crosswalk file.

@jwalsh28 @cdsolari Should the population variable that determines quality be the total population or the population of homeowners?

Reply:
Manu-- did you get a response from Claudia and John? Flagging that I will have to re-check this on a re-review once it has been confirmed.

Reply:
@malcalakovalski @kmartinchek I think the quality flag should be based on the total population. We expect that we will have people of color shut out of homeownership opportunities, and if they make up a large part of a community, but have a small share of the collective home value, that will be a strong quality finding. As opposed to if the group has a small share of the collective home values but they already make up such a small share of the community's population. So, please make quality based on the total population. Thanks!!!!!

Reply:
Will do, thank you!!